src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_VALUE = 20;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private double value;
+
+    public double getRandomValue() {
+        return value = random.nextDouble(MAX_VALUE);
+    }
+
+    public Figure getRandomFigure(Figure[] figures) {
+        int index = random.nextInt(figures.length);
+        String color = colorSupplier.getRandomColor();
+
+        switch (index) {
+            case 0:
+                double radius = getRandomValue();
+                return new Circle(radius, color);
```

Comment:
```suggestion
                return new Circle(getRandomValue(), color);
```

check other figures

Reply:
Ok, I understand this comment and this will be right, but super(color) is not right then

Reply:
Sorry, this code was like that before and you or another mentor said to me to create new variable and send this variable in order to create this code more readable